### PR TITLE
rpc: Adding a 'logpath' entry to getrpcinfo

### PR DESCRIPTION
**Pull request by darosior**
as discussed in #15438

### CODE REVIEW

**Comment by fanquake** [comment]
This is failing on both Windows builds:

[Win32](https://travis-ci.org/bitcoin/bitcoin/jobs/498675434)
```
  GEN      qt/moc_bantablemodel.cpp
rpc/server.cpp: In function ‘UniValue getrpcinfo(const JSONRPCRequest&)’:
rpc/server.cpp:317:72: error: no matching function for call to ‘UniValue::UniValue(UniValue::VType, const value_type*)’
     UniValue log_path(UniValue::VSTR, LogInstance().m_file_path.c_str());
                                                                        ^
In file included from ./rpc/protocol.h:16:0,
                 from ./rpc/server.h:10,
                 from rpc/server.cpp:6:
./univalue/include/univalue.h:46:5: note: candidate: UniValue::UniValue(const char*)
```

[Win64](https://travis-ci.org/bitcoin/bitcoin/jobs/498675435)
```
  GEN      qt/moc_bantablemodel.cpp
  GEN      qt/moc_bitcoinaddressvalidator.cpp
rpc/server.cpp: In function ‘UniValue getrpcinfo(const JSONRPCRequest&)’:
rpc/server.cpp:317:72: error: no matching function for call to ‘UniValue::UniValue(UniValue::VType, const value_type*)’
     UniValue log_path(UniValue::VSTR, LogInstance().m_file_path.c_str());
                                                                        ^
In file included from ./rpc/protocol.h:16:0,
                 from ./rpc/server.h:10,
                 from rpc/server.cpp:6:
./univalue/include/univalue.h:46:5: note: candidate: UniValue::UniValue(const char*)
```

**Comment by darosior** [comment]
I changed `c_str()` to `native()` in order to return a `string_type`.

**Comment by Sjors** [comment]
Concept ACK. Can you add a test, e.g. in `test/functional/interface_rpc.py`? You can get the expected log path with `os.path.join(self.nodes[0].datadir, 'debug.log')`

Using `native()` didn't work. Maybe @ken2812221 knows why the win32 and win64 builds fail?

It's better to remove `fixes #15438` from the commit message. It's already in the pull request description, which automatically gets added to the merge commit.

File: src/rpc/server.cpp [hunk 1]
```diff
@@ -314,6 +314,10 @@ static UniValue getrpcinfo(const JSONRPCRequest& request)
     UniValue result(UniValue::VOBJ);
     result.pushKV("active_commands", active_commands);
 
+    const std::string path = LogInstance().m_file_path.native().c_str();
```
Comment (ken2812221) [inline src/rpc/server.cpp]: Could use `m_file_path.string()` instead. It might fix the issue.

**Comment by darosior** [comment]
@Sjors `os.path.join(self.nodes[0].datadir, 'debug.log')` does not link to the good file. It outputs `/tmp/bitcoin_func_test_tj4444vh/node0/debug.log` instead of `/tmp/bitcoin_func_test_tj4444vh/node0/regtest/debug.log`.

File: src/rpc/server.cpp (hunk 1 above)
Comment (darosior) [reply src/rpc/server.cpp]: Thanks, it fixed it.

File: test/functional/interface_rpc.py [hunk 2]
```diff
@@ -21,6 +22,7 @@ def test_getrpcinfo(self):
         command = info['active_commands'][0]
         assert_equal(command['method'], 'getrpcinfo')
         assert_greater_than_or_equal(command['duration'], 0)
+        assert(info['logpath'] == os.path.join(self.nodes[0].datadir, 'regtest', 'debug.log'))
```
Comment (promag) [inline test/functional/interface_rpc.py]: Use `assert_equal`?

**Review by promag — COMMENTED** [review]
IMO this is not RPC info and as such should not be added here.

File: test/functional/interface_rpc.py (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/interface_rpc.py]: Yes, please. Otherwise a failure can not be debugged

Comment (darosior) [reply test/functional/interface_rpc.py]: Fixed

**Comment by Sjors** [comment]
tACK 59c91b9, but let's wait for discussion in #15438 to settle.

The lack of escape characters (e.g. space) is annoying, but I'm not sure if it's appropriate to add them. 

```sh
bitcoin-cli getrpcinfo | jq -r '.logpath'
/Users/bitcoin/Library/Application Support/Bitcoin/testnet3/debug.log

tail -f "`bitcoin-cli getrpcinfo | jq -r '.logpath'`"
```

**Comment by ryanofsky** [comment]
> The lack of escape characters (e.g. space) is annoying, but I'm not sure if it's appropriate to add them.

What kind of escaping would you want to add here? UniValue already adds JSON escaping, so I would think any extra layers of escaping would just just break your tail command and be more annoying.

**Comment by Sjors** [comment]
@ryanofsky only escaping spaces in the path

**Comment by promag** [comment]
NACK, I think #15493 is a better solution for OP.

**Comment by laanwj** [comment]
> What kind of escaping would you want to add here? UniValue already adds JSON escaping, so I would think any extra layers of escaping would just break your tail command and be more annoying.

Definite *don't* add any extra escaping. If we add this field, it needs to be usable from other JSON clients as well which don't do any other unquoting besides parsing JSON. 
String sanitization is only relevant for potentially hostile input, from, say, the network.

**Comment by darosior** [comment]
Rebased, but is it still needed or #15493 will be prefered ?

**Comment by laanwj** [comment]
I think getting the log path through RPC can be useful (say, if you have some kind of monitoring program such as [bitcoind-ncurses](https://github.com/esotericnonsense/bitcoind-ncurses2) that connects to the RPC and needs to show the logs) and  #15493 doesn't entirely replace that use case.

Tested ACK 8a6810d0d2759c69f63b53c48aa79e0cfdd88ffb

**Pull request merged**